src/ProofGenerator.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Web3 from 'web3';
+import * as Web3Utils from 'web3-utils';
+import { GetProof } from 'web3-eth';
+
+const RLP = require('rlp');
+
+// This is the position of message outbox defined in GatewayBase.sol
+const MESSAGE_OUTBOX_OFFSET = '9';
```

Comment:
It will become dependent on mosaic-contract npm package.
This is just a default value as per the latest Gateway contracts.
The caller can pass the proper offset as param in `getOutboxProof` and `getInboxProof`